pandas/tools/tile.py

```diff
@@ -185,6 +185,13 @@ def qcut(x, q, labels=None, retbins=False, precision=3, duplicates='raise'):
     x, dtype = _coerce_to_type(x)
 
     if is_integer(q):
+        if x.size == 0:
+            raise ValueError('Cannot qcut empty array')
+
+        rng = (nanops.nanmin(x), nanops.nanmax(x))
+        if rng[0] == rng[1] and q == 1:
```

Comment:
The only reason it could be useful is to keep the number of quantiles as per user request. This is the only difference compare to 'drop'. I understand this might not be super useful, so it's up to you. Still I would keep the option for internal use.